regolith/helpers/a_proposalhelper.py

```diff
@@ -0,0 +1,197 @@
+                                                 "to be stated on the current and pending form"
+                       )
+    subpi.add_argument("--months_summer", help="Number of working months in the summer to be"
+                                                "stated on the current and pending form"
+                       )
+    subpi.add_argument("--scope", help="Scope of the project"
+                       )
+    subpi.add_argument("-f", "--funder",
+                       help="Agency where the proposal is being submitted"
+                       )
+    subpi.add_argument("-n", "--notes", nargs="+",
+                       help="Anything to note"
+                       )
+    return subpi
+
+
+class ProposalAdderHelper(DbHelperBase):
+    """Helper for adding a proposal to the proposals collection.
+
+       A proposal is a dictionary object describing a research or
+        project proposal submitted by the group.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = "a_proposal"
+    needed_dbs = [f'{TARGET_COLL}', 'people']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        rc.pi_id = get_pi_id(rc)
+        rc.coll = f"{TARGET_COLL}"
+        if not rc.database:
+            rc.database = rc.databases[0]["name"]
+        gtx[rc.coll] = sorted(
+            all_docs_from_collection(rc.client, rc.coll), key=_id_key
+        )
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def db_updater(self):
+        gtx = self.gtx
+        rc = self.rc
+        if not rc.date:
+            now = dt.date.today()
+        else:
+            now = date_parser.parse(rc.date).date()
+        key = f"{str(now.year)[2:]}_{''.join(rc.name.casefold().split()).strip()}"
+
+        coll = self.gtx[rc.coll]
+        pdocl = list(filter(lambda doc: doc["_id"] == key, coll))
+        if len(pdocl) > 0:
+            raise RuntimeError(
+                "This entry appears to already exist in the collection")
+        else:
+            pdoc = {}
+        pdoc.update({'_id': key})
+        if rc.amount:
+            pdoc.update({'amount': rc.amount})
+        else:
+            pdoc.update({'amount': ''})
+        if rc.authors:
+            if isinstance(rc.authors, str):
+                pdoc.update({'authors': [rc.authors]})
+            else:
+                pdoc.update({'authors': rc.authors})
+        else:
+            pdoc.update({'authors': ['']})
+        if rc.begin_date:
+            pdoc.update({'begin_date': rc.begin_date})
+        else:
+            pdoc.update({'begin_date': ''})
+        cpp_info = {}
+        if rc.cppflag:
+            cpp_info['cppflag'] = rc.cppflag
+        if rc.other_agencies:
+            cpp_info['other_agencies_submitted'] = rc.other_agencies
```

Comment:
These all need default values, but as discussed above, experiment with not changing code here but using defaults above.  Same comments apply below.

Reply:
done